src/main/java/com/spring_boot/repository/MailDao.java

```diff
@@ -0,0 +1,95 @@
+package com.spring_boot.repository;
+
+import com.home_work.spring_boot.entity.Letter;
+import com.home_work.spring_boot.services.PropertiesService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+import java.io.*;
+import java.util.LinkedList;
+import java.util.Map;
+
+@Repository
+public class MailDao {
```

Comment:
Consider separating this file-based implementation from other components via interface. Then everywhere in the code that depends on it use an interface(just define dependency as an interface. spring will inject proper implementation for you). This will help you to switch from file-based approach to the database in future with minimal changes to the code